src/main/webapp/placeGuides/ListPlaceGuideDisplayer.js

```diff
@@ -2,75 +2,132 @@
  * This class is responsible for representing
  * the placeGuides on a scrollable list.
  */
-class ListPlaceGuideDisplayer {
-  constructor() {
-    this._placeGuidesOnList = {};
-  }
 
-  update(placeGuides) {
-    this.removePreviousPlaceGuidesFromList(placeGuides);
-    this.addNewPlaceGuidesToList(placeGuides);
-  }
+class ListPlaceGuideDisplayer {
 
-  removeFromList(placeGuideId) {
+  static QUERY = {
+    "ALL_PUBLIC_IN_MAP_AREA": {
+      listTitle: "Discover Guides",
+      listSubTitle: "in selected map area"
+    },
+    "CREATED_ALL_IN_MAP_AREA": {
+      listTitle: "My Guides",
+      listSubTitle: "in selected map area"
+    },
+    "BOOKMARKED": {
+      listTitle: "Bookmarked Guides",
+      listSubTitle: ""    }
+  };
 
+  constructor(placeGuideDisplayQuery) {    
+    this._listPlaceGuideDisplayerDiv = document.getElementById("listPlaceGuideDisplayer");
+    this._listPlaceGuideDisplayerDiv.classList.add(
+        "list-group", "my-place-guide-list", "list-place-guide-displayer", "form-card");
+    this._placeGuideDisplayQuery = placeGuideDisplayQuery;
+    this._listPlaceGuideDisplayerDiv.appendChild(
+        this.createListTitle(
+            ListPlaceGuideDisplayer.QUERY[placeGuideDisplayQuery]));
   }
 
-  addToList(placeGuideId) {
+  get listPlaceGuideDisplayerDiv() {
+    return this._listPlaceGuideDisplayerDiv;
+  }
 
+  update(placeGuides) {
+    this.removeAllPlaceGuidesFromList();
+    this._listPlaceGuideDisplayerDiv.appendChild(
+        this.createListTitle(
+            ListPlaceGuideDisplayer.QUERY[this._placeGuideDisplayQuery]));
+    this.addPlaceGuidesToList(placeGuides);
```

Comment:
Consider removing only the ones which don't need to be added again. And then add only the ones which were not there previously.
For a larger number of placeguides I think it may significantly increase the efficiency. Plus, it's a little annoying that whenever the map bounds are changed, the order of placeguides on the list is completely changed. Instead, We could just append the new ones at the end.

Reply:
modified the update method.